### PR TITLE
Fix mob pull issue

### PR DESCRIPTION
**Pull request by Anzarot121**
## About the PR

Once again allows creatures that inherit from simplemob(hamster, slime, spider, etc) to pull items.
## Why / Balance

fix issue https://github.com/space-wizards/space-station-14/issues/22465
## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

None
**Changelog**
:cl:
- fix: Fixed a problem in the code due to which mobs, including hamsters, slimes, spiders, xenos, could not pull things.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
I feel like this really shouldnt be on simplemob. Entities should just add this as needed because every mob being able to drag can be highly problematic.

**Comment by metalgearsloth** [comment]
Wasn't the point to stop them pulling and breaking / stealing things?

**Comment by Anzarot121** [comment]
> I feel like this really shouldnt be on simplemob. Entities should just add this as needed because every mob being able to drag can be highly problematic.

Well, this is how it was before the PR that nerfed mothroaches.

**Comment by Anzarot121** [comment]
> Wasn't the point to stop them pulling and breaking / stealing things?

Only mothroaches. But for some reason the author of that PR forbade all creatures that inherit from simplemob from doing this. I added a line of code that saves his changes but again allows everyone else to drag things along.

**Review by Errant-4 — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 1]
```diff
@@ -443,7 +443,8 @@
     prototype: Mothroach
   - type: TypingIndicator
     proto: moth
-
+  - type: Puller
+    needsHands: true
 - type: entity
```
[the remark this answers is not in the document]
Comment (Errant-4) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:449]: That's almost right, but it starts with `- type: entity` , so what you have done is, you visually dislodged the end of the mothroach codeblock. Observe where the indentation changes

[the remark this answers is not in the document]
Comment (Anzarot121) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:449]: done

**Review by Errant-4 — COMMENTED** [review]
Looks good  in regards to what the original concept of the PR was. I would clarify the changelog entry a bit, and end it with a "."

Now the question is, do you want to call it a day and leave it like this, which would be fair. Or consider emogarbage's suggestion, which I will admit is probably a better idea in the long run. Slightly awkward, because it involves doing the very first approach you tried to do, and I told you not to do

**Comment by Anzarot121** [comment]
> Looks good in regards to what the original concept of the PR was. I would clarify the changelog entry a bit, and end it with a "."
> 
> Now the question is, do you want to call it a day and leave it like this, which would be fair. Or consider emogarbage's suggestion, which I will admit is probably a better idea in the long run.

I believe that for my first PR, what I have now will be enough. I spent 4 hours on this, perhaps Emo’s proposal needs to be implemented separately from this PR.

**Comment by Anzarot121** [comment]
I also think there needs to be a discussion about which existing mobs should retain the ability to pull things, which should be added this ability, and which should be removed.

**Comment by Anzarot121** [comment]
So?

**Comment by OctoRocket** [comment]
This would certainly nice to have back.

**Comment by metalgearsloth** [comment]
> > Wasn't the point to stop them pulling and breaking / stealing things?
> 
> Only mothroaches. But for some reason the author of that PR forbade all creatures that inherit from simplemob from doing this. I added a line of code that saves his changes but again allows everyone else to drag things along.

Mothroaches are a symptom of the problem, hamsters or whatever billion simplemobs there are being able to do it is still an issue.

**Comment by OctoRocket** [comment]
What was the original issue with moth roaches being able to pull things anyways?

**Comment by gardspirito** [comment]
Updates?

**Comment by metalgearsloth** [comment]
> What was the original issue with mothroaches being able to pull things anyways?

Mobs that can go under doors pulling things.

**Comment by K-Dynamic** [comment]
> Mobs that can go under doors pulling things.

I thought items don't slide under doors even when pulled

Mobs can pull items through plastic flaps though

**Comment by metalgearsloth** [comment]
People asking for updates when the original request is still pending.

**Comment by Anzarot121** [comment]
Oh okay. Well. @EmoGarbage404, Do you have a rough list of mobs or groups of mobs that should be allowed to pull?

**Pull request merged**